pkg/ingester/ingester.go

```diff
@@ -2881,6 +2881,64 @@ func (i *Ingester) flushHandler(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+const (
+	mode     = "mode"
+	READONLY = "READONLY"
+	ACTIVE   = "ACTIVE"
+)
+
+func (i *Ingester) ModeHandler(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseForm()
+	if err != nil {
```

Comment:
does this ever happens?. I mean makes sense what you did but I don't see a test for it

Reply:
I agree that this seems pretty uncommon, but it is mostly checking if it is a valid request. Looking at the inner "ParseForm" it can happen with wrong media types for POST, or malformatted query params. 
We do have checks for "ParseForm" when we use it. I feel it is safer to do the check. 
Added test